ui/components/app/snaps/snap-connect-cell/snap-connect-cell.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Box from '../../../ui/box';
+import {
+  IconColor,
+  AlignItems,
+  Display,
+  FontWeight,
+} from '../../../../helpers/constants/design-system';
+import { getSnapName } from '../../../../helpers/utils/util';
+import {
+  Icon,
+  IconName,
+  IconSize,
+  Text,
+  ValidTag,
+} from '../../../component-library';
+import Tooltip from '../../../ui/tooltip/tooltip';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import SnapAvatar from '../snap-avatar/snap-avatar';
+
+export default function SnapConnectCell({ origin, snapId }) {
+  const t = useI18nContext();
+  const friendlyName = getSnapName(snapId);
```

Comment:
same here, might be good to get `subjectMetadata` of the snap in case we have the name in the state.